httprule/client.go

```diff
@@ -43,6 +95,13 @@ func (c *ClientConn) Invoke(ctx context.Context, method string, args, reply inte
 	if err != nil {
 		return err
 	}
+	if c.header != nil {
```

Comment:
I think this if is not needed: https://play.golang.com/p/akjM13n9f20

Reply:
you are right this is not needed. I remember saying this a couple of weeks ago (you can read from a nil map). Will fix.